### PR TITLE
Add call_external functions to derive_fn macro

### LINKED ISSUE #243: Test Fwk: Generate another wrap method that calls the contract fn via the env.invoke_function (macros)

### PR DESCRIPTION
**Pull request by jonjove**
Resolves #243

### CODE REVIEW

File: macros/src/derive_fn.rs
Comment (leighmcculloch) [inline macros/src/derive_fn.rs:93]: It helps with readability in the long tuple (too long!) if we name the args here.
```suggestion
                let invoke_call = quote! { #ident };
                (spec, arg, call, invoke_arg, invoke_call)
```

File: macros/src/derive_fn.rs
Comment (leighmcculloch) [inline macros/src/derive_fn.rs:183]: We don't need to resolve this immediately right now before merging, but we need to come up with a good naming strategy for the different ways we call things, because I think "external" is ambiguous. It's possible we can't do better than this though. I at least can't think of anything better in this moment.

File: macros/src/derive_fn.rs
Comment (leighmcculloch) [inline macros/src/derive_fn.rs:193]: We might want to return the `Result` given that the function will be used in testing, there's a good chance people will be asserting on the return value, and it is easy to assert on a `Result` which makes testing a bit clearer imo.

**Review by leighmcculloch — APPROVED** [review]
👏🏻 This is wonderful. Left comments, but none need to block merging this.

**Comment by leighmcculloch** [comment]
@jonjove Looks like there is a build error. I don't think it was the commit I pushed as the original PR commit failed too.

File: macros/src/derive_fn.rs
Comment (jonjove) [reply macros/src/derive_fn.rs:185]: Opened #329

File: macros/src/derive_fn.rs
Comment (jonjove) [reply macros/src/derive_fn.rs:195]: Opened #330

**Pull request merged**
